tomb: 2.9 -> 2.10

anthonyroussel opened the pull request:
## Description of changes

https://github.com/dyne/Tomb/blob/master/ChangeLog.md#210
https://github.com/dyne/Tomb/compare/v2.9...v2.10

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

anthonyroussel committed:
tomb: add passthru.updateScript

anthonyroussel committed:
tomb: add anthonyroussel to maintainers

cafkafk commented on pkgs/os-specific/linux/tomb/default.nix:
```diff
@@ -1,16 +1,18 @@
 { stdenv, lib, fetchFromGitHub, makeWrapper
-, gettext, zsh, pinentry, cryptsetup, gnupg, util-linux, e2fsprogs, sudo
+, gettext, zsh, pinentry, cryptsetup, gnupg, libargon2, util-linux, e2fsprogs, sudo
+, testers
+, nix-update-script
 }:
```
The formatting here should probably be more consistent

anthonyroussel replied:
Indeed, thanks for the review. I applied formatting on this file using `nixpkgs-fmt`.

bjornfor commented on pkgs/os-specific/linux/tomb/default.nix:
```diff
@@ -31,12 +31,13 @@ stdenv.mkDerivation rec {
 
     wrapProgram $out/bin/tomb \
-      --prefix PATH : $out/bin:${lib.makeBinPath [ cryptsetup gettext gnupg pinentry util-linux e2fsprogs ]}
+      --prefix PATH : $out/bin:${lib.makeBinPath [ cryptsetup gettext gnupg libargon2 pinentry sudo util-linux e2fsprogs ]}
```
How does this sudo work? I think sudo generally has to be a setuid binary provided by the system (`/run/wrappers/bin/sudo` on NixOS).

anthonyroussel replied:
@bjornfor Hello, thanks for the review.

Yes indeed, I modified the package to use `/run/wrappers/bin/sudo` for sudo (which is also overridable through `sudoPath`).

I also had to patch the package because tomb calls sudo through $PATH and does not support using a specific binary path for sudo command.

What do you think?

anthonyroussel committed:
tomb: format with nixpkgs-fmt

anthonyroussel committed:
tomb: 2.9 -> 2.10

https://github.com/dyne/Tomb/blob/master/ChangeLog.md#210
https://github.com/dyne/Tomb/compare/v2.9...v2.10

nixos-discourse commented:
This pull request has been mentioned on **NixOS Discourse**. There might be relevant details there:

https://discourse.nixos.org/t/prs-ready-for-review/3032/2821

pbsds commented on pkgs/os-specific/linux/tomb/default.nix:
```diff
@@ -1,29 +1,42 @@
-, sudo
+, testers
+, nix-update-script
+, sudoPath ? "/run/wrappers/bin/sudo"
```
A problem with this approach is that it does not work outside of nixos.

anthonyroussel replied:
@pbsds Hello. Thanks for the review.
Do you have any ideas about the best way to solve this problem?

I don't have any other options apart from allowing the user to override the `sudoPath` attribute according to their system.

An another option could be also to remove this `sudoPath` attribute and the sudo buildInput, but this would make installation of the `tomb` package conditional on the presence of the sudo binary in its PATH.

Do you have any idea?

pbsds replied:
Perhaps a `pkgs.writeShellScript` replacement? A simple script that checks for the existence of the nixos sudo path then tries to use it, and otherwise falls back to the sudo in PATH.

bjornfor replied:
Why not always use PATH?

anthonyroussel replied:
Okay thanks for the review, I changed my PR to use the sudo binary in the PATH instead.

pbsds commented on pkgs/os-specific/linux/tomb/fix-sudo-path.patch:
```diff
@@ -0,0 +1,46 @@
+ 		[[ -n "$SUDO_ASKPASS" ]] && local sudo_askpass="--askpass"
+# Calls the sudo binary path specified in $SUDO
+-		sudo $sudo_askpass -p "
++		$SUDO $sudo_askpass -p "
```
This looks like it could be submitted upstream

anthonyroussel committed:
tomb: add all required dependencies with wrapProgram

pbsds commented on pkgs/os-specific/linux/tomb/default.nix:
```diff
@@ -23,22 +41,39 @@ stdenv.mkDerivation rec {
-    platforms   = platforms.linux;
+    homepage = "https://www.dyne.org/software/tomb/";
+    changelog = "https://github.com/dyne/Tomb/blob/v${version}/ChangeLog.md";
+    license = licenses.gpl3;
```
```suggestion
    license = licenses.gpl3Only;
```

or `gpl3Plus`, depending on what upstream says. [Please view this explanation](https://github.com/jtojnar/nixpkgs-hammering/blob/main/explanations/unclear-gpl.md)

anthonyroussel replied:
Thanks! I fixed the meta.license to gpl3Only

pbsds reviewed (approved):
Looking very good. A final nit :)

anthonyroussel committed:
tomb: fix meta.license

pbsds commented:
Thanks!
